src/Features/Core/Portable/CodeFixes/Configuration/ConfigurationUpdater.cs

```diff
@@ -294,31 +299,40 @@ private AnalyzerConfigDocument FindOrGenerateEditorConfig()
                 }
             }
 
-            return ImmutableArray<(OptionKey, ICodeStyleOption, IEditorConfigStorageLocation2)>.Empty;
+            return ImmutableArray<(OptionKey, ICodeStyleOption, IEditorConfigStorageLocation2, bool)>.Empty;
         }
 
-        private SourceText GetNewAnalyzerConfigDocumentText(
-            SourceText originalText,
-            Dictionary<string, TextLine> headers)
+        private SourceText GetNewAnalyzerConfigDocumentText(SourceText originalText)
         {
             // Check if an entry to configure the rule severity already exists in the .editorconfig file.
             // If it does, we update the existing entry with the new severity.
-            var configureExistingRuleText = CheckIfRuleExistsAndReplaceInFile(originalText, headers);
-            if (configureExistingRuleText != null)
+            var configureExistingRuleTextAndValidHeader = CheckIfRuleExistsAndReplaceInFile(originalText);
+            if (configureExistingRuleTextAndValidHeader.Item1 != null)
             {
-                return configureExistingRuleText;
+                return configureExistingRuleTextAndValidHeader.Item1;
             }
 
             // We did not find any existing entry in the in the .editorconfig file to configure rule severity.
             // So we add a new configuration entry to the .editorconfig file.
-            return AddMissingRule(originalText, headers);
+            return AddMissingRule(originalText, configureExistingRuleTextAndValidHeader.Item2, configureExistingRuleTextAndValidHeader.Item3);
         }
 
-        private SourceText CheckIfRuleExistsAndReplaceInFile(
-            SourceText result,
-            Dictionary<string, TextLine> headers)
+        private Tuple<SourceText, TextLine?, TextLine?> CheckIfRuleExistsAndReplaceInFile(SourceText result)
         {
-            string mostRecentHeader = null;
+            // Finds the relative path between editorconfig directory and diagnostic filepath.
+            var editorConfigDirectory = PathUtilities.GetDirectoryName(FindOrGenerateEditorConfig().FilePath).ToLowerInvariant();
```

Comment:
Instead of invoking `FindOrGenerateEditorConfig()` again here, can we pass in the `editorConfigDocument` computed [here](https://github.com/dotnet/roslyn/blob/master/src/Features/Core/Portable/CodeFixes/Configuration/ConfigurationUpdater.cs#L168) down to `GetNewAnalyzerConfigDocumentText` and then `CheckIfRuleExistsAndReplaceInFile`?